.github/workflows/checks.yaml

```diff
@@ -0,0 +1,82 @@
+name: Checks
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  ci: 
+    runs-on: ubuntu-latest
+    outputs:
+      status: ${{ job.status }}
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name : Run EC Validate (keyless)
+      uses: ./
+      with:
+        image: ghcr.io/enterprise-contract/golden-container:latest
+        identity: https:\/\/github\.com\/(slsa-framework\/slsa-github-generator|enterprise-contract\/golden-container)\/
+        issuer: https://token.actions.githubusercontent.com
+
+    # - name : Run EC Validate (Long_Lived)
+    #   uses: ./
+    #   with:
+    #     image: quay.io/redhat-appstudio/ec-golden-image:latest
+    #     key: ${{ vars.PUBLIC_KEY }}
+    #     policy: " " #TODO Ignore until image is fixed 
+    #     extra-params: --ignore-rekor
+
+
+  release:
+    runs-on: ubuntu-latest
+    needs: ci
+    if: needs.ci.outputs.status == 'success' && github.ref == 'refs/heads/main' && github.event_name != 'pull_request'
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Setup GitHub Auth for gh cli
+      run: echo ${{ secrets.GHTOKEN }} | gh auth login --with-token
```

Comment:
I don't think we need to do this. According to the [docs](https://docs.github.com/en/actions/using-workflows/using-github-cli-in-workflows), you should be able to just set the environment variable `GITHUB_TOKEN` and gh-cli should pick it up.

The value used should probably be the one from `secrets.GITHUB_TOKEN` which is automatically created by GitHub. We don't have to manage that secret.

You may have to play with the job permissions to make sure it has access to do the needed operations.

Reply:
This comment was resolved without being addressed. We should either remove the step named "Setup GitHub Auth for gh cli" or explain why it is needed.

Reply:
Apologies for the oversight; I overlooked the 'push' post-click🧑‍🎓. Upon testing, we don't need permissions, such as `contents: write`, I got rid of the extra step and n used a environment variable instead like you suggested.